github-image-preview.user.js

```diff
@@ -215,28 +216,16 @@
 				content = updateTemplate(url, `${title()}${svgPlaceholder(url)}`);
 			} else {
 				// *** non-images (file/folder icons) ***
-				const svg = $("svg", row);
+				const svg = $("[role='gridcell'] svg, [role='gridcell'] img", row);
 				if (svg) {
 					// non-files svg class: "directory", "submodule" or "symlink"
 					// add "ghip-folder" class for file-filters userscript
 					const noExt = svg.matches(folderIconClasses) ? " ghip-folder" : "";
-					// add xmlns otherwise the svg won't work inside an img
-					// GitHub doesn't include this attribute on any svg octicons
-					const svgHTML = svg.outerHTML.replace("<svg", "<svg xmlns='http://www.w3.org/2000/svg'");
+					const clone = svg.cloneNode(true);
+                    clone.classList.add("ghip-non-image");
 					// include "leaflet-tile-container" to invert icon for GitHub-Dark
 					content = `${title("non-image")}<span class="leaflet-tile-container${noExt}">` +
-						`<img class="ghip-non-image" src="data:image/svg+xml;base64,` +
-						window.btoa(svgHTML) + `"/>`;
-					// get file name + extension
-					const str = url.substring(url.lastIndexOf("/") + 1, url.length);
-					// don't include extension for folders, or files with no extension,
-					// or files starting with a "." (e.g. ".gitignore")
-					content += (!noExt && str.indexOf(".") > 0) ?
-						"<h4 class='ghip-file-type'>" +
-						str
-							.substring(str.lastIndexOf(".") + 1, str.length)
-							.toUpperCase() +
-						"</h4></span>" : "</span>";
```

Comment:
This does not look good with file icon extensions, so I dropped it. However, it makes it a little worse for those who _don't_ use them. I believe the tradeoff is worth it, let me know your thoughts.

Reply:
Yeah, I think you're right.. I'm going to go ahead and merge your work, but I'm going to tweak it a bit to show the full file name. Thanks again!